### PR TITLE
Updating test scripts

### PR DESCRIPTION
**Pull request by manika137**
### Description of PR
Updating test scripts.

HNS-OAuth-DFS
HNS-SharedKey-DFS
NonHNS-SharedKey-DFS
AppendBlob-HNS-OAuth-DFS
NonHNS-SharedKey-Blob
NonHNS-OAuth-DFS
NonHNS-OAuth-Blob
AppendBlob-NonHNS-OAuth-Blob
HNS-Oauth-DFS-IngressBlob
NonHNS-Oauth-DFS-IngressBlob

### CODE REVIEW

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 1]
```diff
@@ -126,39 +177,95 @@ do
...
+         runNonHNSSharedKeyDFSTest
          break
          ;;
-       AppendBlob-HNS-OAuth)
-         runAppendBlobHNSOAuthTest
+       AppendBlob-HNS-OAuth-DFS)
+         runAppendBlobHNSOAuthDFSTest
          break
          ;;
+       NonHNS-SharedKey-Blob)
```
Comment (anmolanmol1234) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:204]: indentation is not proper

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 2]
```diff
@@ -126,39 +177,95 @@ do
...
+        runNonHNSSharedKeyDFSTest
+        runAppendBlobHNSOAuthDFSTest
+        runNonHNSSharedKeyBlobTest
+        runNonHNSOAuthDFSTest
+        runNonHNSOAuthBlobTest
+        runAppendBlobNonHNSOAuthBlobTest
+        runHNSOAuthDFSIngressBlobTest
+        runNonHNSOAuthDFSIngressBlobTest
+
+#         runHNSOAuthTest
```
Comment (anmolanmol1234) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh]: can remove the comments

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 3]
```diff
@@ -32,36 +32,87 @@ processCount=8
...
+  triggerRun "AppendBlob-HNS-OAuth-DFS" "$accountName" "$runTest" $processCount "$cleanUpTestContainers"
+}
+
+runNonHNSSharedKeyBlobTest()
+{
+  accountName=$(xmlstarlet sel -t -v '//property[name = "fs.azure.nonHnsTestAccountName"]/value' -n $azureTestXmlPath)
+  fnsBlobConfigFileCheck "$accountName"
+  PROPERTIES=("fs.azure.account.auth.type")
+  VALUES=("SharedKey")
+  triggerRun "NonHNS-SharedKey-Blob" "${accountName}_blob" "$runTest" $processCount "$cleanUpTestContainers"
```
Comment (anmolanmol1234) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:73]: Does ${accountName}_blob works ?

**Review by anmolanmol1234 — CHANGES_REQUESTED** [review]

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 4]
```diff
@@ -32,36 +32,87 @@ processCount=8
...
+runNonHNSOAuthBlobTest()
+{
+  accountName=$(xmlstarlet sel -t -v '//property[name = "fs.azure.nonHnsTestAccountName"]/value' -n $azureTestXmlPath)
+  fnsBlobConfigFileCheck "$accountName"
+  PROPERTIES=("fs.azure.account.auth.type")
+  VALUES=("OAuth")
+  triggerRun "NonHNS-OAuth-Blob" "${accountName}_blob" "$runTest" $processCount "$cleanUpTestContainers"
+}
+
+runAppendBlobNonHNSOAuthBlobTest()
```
Comment (saxenapranav) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:93]: Should we have for dfs endpoint as well.

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 5]
```diff
@@ -32,36 +32,87 @@ processCount=8
...
+{
+  accountName=$(xmlstarlet sel -t -v '//property[name = "fs.azure.nonHnsTestAccountName"]/value' -n $azureTestXmlPath)
+  fnsBlobConfigFileCheck "$accountName"
+  PROPERTIES=("fs.azure.account.auth.type" "fs.azure.test.appendblob.enabled")
+  VALUES=("OAuth" "true")
+  triggerRun "AppendBlob-NonHNS-OAuth-Blob" "${accountName}_blob" "$runTest" $processCount "$cleanUpTestContainers"
+}
+
+runHNSOAuthDFSIngressBlobTest()
```
Comment (saxenapranav) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:102]: Blob ingress on an hns account, dont think it should happen. @anmolanmol1234, what you feel.

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 6]
```diff
@@ -32,36 +32,87 @@ processCount=8
...
+{
+  accountName=$(xmlstarlet sel -t -v '//property[name = "fs.azure.hnsTestAccountName"]/value' -n $azureTestXmlPath)
+  PROPERTIES=("fs.azure.account.auth.type" "fs.azure.ingress.service.type")
+  VALUES=("OAuth" "blob")
```
Comment (saxenapranav) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:106]: I think this should be DFS or BLOB. Not in the uppercase. Reason being, in the code, it resolves from enum AbfsServiceType which has the values in uppercase.

**Review by saxenapranav — CHANGES_REQUESTED** [review]
Congrats on the first PR @manika137, thanks for the patch. Great code. Some thoughts.

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 5 above)
Comment (anmolanmol1234) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:102]: This should work @saxenapranav. We have added support to handle it

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh [hunk 7]
```diff
@@ -126,39 +177,95 @@ do
    esac
 done
 
+## SECTION: CHECK FOR FNS+BLOB XML FILE
```
Comment (anujmodi2021) [inline hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh]: Move this section and function to testsupport.sh along with other utility functions

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 1 above)
Comment (manika137) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:204]: Corrected

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 6 above)
Comment (anmolanmol1234) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:106]: I think Anuj has added support for it being case insensitive, right @anujmodi2021 ?

**Review by anujmodi2021 — CHANGES_REQUESTED** [review]
Good Work with the patch.
Added a change request.

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 3 above)
Comment (anujmodi2021) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:73]: Seems to be working...

Comment (manika137) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:73]: Yes, it's creating new file if not already created.
It's showing the same file name in abfs-test-combinations-config.xml for the required combo

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 6 above)
Comment (manika137) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:106]: Yes, it is case insensitive

Comment (anujmodi2021) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:106]: This was supposed to be case insensitive but looks like it's not.
May be that's why ingress related tests are failing.

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 7 above)
Comment (manika137) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh]: Right, makes sense

File: hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh (hunk 6 above)
Comment (anujmodi2021) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:106]: I will fix this and make it case insensitive, meanwhile @manika137 can you trigger a ingressBlob combination after changing this setting value from blob to BLOB. to confirm this hypothesis.

Comment (manika137) [reply hadoop-tools/hadoop-azure/dev-support/testrun-scripts/runtests.sh:106]: Worked

**Review by anujmodi2021 — APPROVED** [review]

**Review by anmolanmol1234 — APPROVED** [review]

**Comment by anmolanmol1234** [comment]
+1 Looks good. Thanks for taking the changes.

**Review by saxenapranav — APPROVED** [review]
+1. Thanks for the changes!

**Pull request merged**